src/main/java/org/openrewrite/staticanalysis/ReplaceWeekYearWithYear.java

```diff
@@ -0,0 +1,77 @@
+package org.openrewrite.staticanalysis;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Markers;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static org.openrewrite.Tree.randomId;
+
+public class ReplaceWeekYearWithYear extends Recipe {
+    private static final MethodMatcher DATE_FORMAT_MATCHER = new MethodMatcher("java.text.SimpleDateFormat <constructor>(..)");
```

Comment:
Note that this will match to three or four different constructors! You'd need to handle each of those in the recipe below.